internal/core/diagnostics.go

```diff
@@ -75,17 +75,17 @@ func (a *Agent) DiagnosticText() string {
 
 }
 
-func (a *Agent) serveInternalMetrics(socketPath string) error {
-	if a.internalMetricsSocketStop != nil {
-		a.internalMetricsSocketStop()
+func (a *Agent) serveInternalMetrics(path string) error {
+	if a.internalMetricsServerStop != nil {
+		a.internalMetricsServerStop()
 	}
 
-	if err := os.MkdirAll(filepath.Dir(socketPath), 0700); err != nil {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
```

Comment:
What does this do on Windows with a pipe path?

Reply:
yeah actually i don't think I should do that on windows.  It explains some behavior I've been trying to understand through their doc on named pipes.  When I create the named pipe I see a directory called "pipe" created at the root of C:\.  I'll work around that on windows.